ambry-protocol/src/main/java/com/github/ambry/protocol/AmbryRequests.java

```diff
@@ -196,6 +196,77 @@ public void handleRequests(NetworkRequest networkRequest) throws InterruptedExce
     }
   }
 
+  @Override
+  public void dropRequest(NetworkRequest request) throws InterruptedException {
+    try {
+      InputStream is = request.getInputStream();
+      DataInputStream dis = is instanceof DataInputStream ? (DataInputStream) is : new DataInputStream(is);
+      RequestOrResponseType type = RequestOrResponseType.values()[dis.readShort()];
+      Response response;
+      long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+      RequestMetricsUpdater metricsUpdater = new RequestMetricsUpdater(requestQueueTime, 0, 0, 0, true);
+      switch (type) {
+        case PutRequest:
+          PutRequest putRequest = PutRequest.readFrom(dis, clusterMap);
+          putRequest.accept(metricsUpdater);
```

Comment:
why not just metricsUpdater.accept here ? 
also, the accept doesn't really accept, kind of a misnomer